crates/turbo-tasks-memory/src/output.rs

```diff
@@ -56,7 +56,17 @@ impl Output {
 
     pub fn link(&mut self, target: RawVc, turbo_tasks: &dyn TurboTasksBackendApi<MemoryBackend>) {
         debug_assert!(*self != target);
-        self.assign(OutputContent::Link(target), turbo_tasks)
+        if let OutputContent::Link(old_target) = &self.content {
+            if *old_target == target {
+                // unchanged
+                return;
```

Comment:
We can early return when returning the same value again.

Previously this notified all dependent tasks, which is unnecessary.